### PR TITLE
fix[ hubspot ] :: simplifying lookup field operation

### PR DESCRIPTION
**Pull request by shrouti1507**
## Description of the change

> Lookup field was double-dynamic.
It was like:
webapp config name : _lookupField_
```
messgae.traits : {
"lookupField" : "email"
"email" : "ABC@gmail.com"
}
```

but now, it will operate in a straight forward manner:

webapp config name : _lookupField_
```
messgae.traits : {
"lookupField" : "ABC@gmail.com"
}
```
## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: v0/destinations/hs/util.js [hunk 1]
```diff
@@ -238,26 +238,14 @@ const searchContacts = async (message, destination) => {
...
     getFieldValueFromMessage(message, propertyName) ||
     traits[`${propertyName}`];
 
-  if (!value) {
-    throw new CustomError(
-      `[HS] Identify:: '${propertyName}' lookup field for contact lookup not found in traits`,
-      400
-    );
-  }
-
```
Comment (utsabc) [inline v0/destinations/hs/util.js:260]: If value is not present how would we handle?

Comment (utsabc) [reply v0/destinations/hs/util.js:260]: Why not use this approach
```
function getLookupFieldValue(message, lookupField) {
      SOURCE_KEYS = ["traits", "context.traits", "properties"]
      value = getValueFromMessage(message, lookUpField);
      if (!value) {
        // Check in free-flowing object level
        SOURCE_KEYS.some(sourceKey => {
          value = getMappingFieldValueFormMessage(
            message,
            sourceKey,
            lookUpField
          );
          if (value) {
            return true;
          }
          return false;
        });
      }
   return value;
}
     
value =  getLookupFieldValue(message, lookupField) || getLookupFieldValue(message, "email");
if(!value) {
  error out
}


```

**Review by utsabc — CHANGES_REQUESTED** [review]

File: v0/destinations/hs/util.js (hunk 1 above)
Comment (shrouti1507) [reply v0/destinations/hs/util.js]: addressed

File: v0/destinations/hs/util.js [hunk 2]
```diff
@@ -205,6 +206,42 @@ const getEmailAndUpdatedProps = properties => {
...
 
+/**
+ * @param {*} message The entire message object
+ * @param {*} sourceKey the base object to search the lookup value from
+ * @param {*} lookupField destination.Config.lookupField or email
+ * @returns returns the lookup value
+ */
+const getMappingFieldValueFormMessage = (message, sourceKey, lookupField) => {
+  const baseObject = get(message, `${sourceKey}`);
+  const lookupValue = baseObject ? baseObject[`${lookupField}`] : null;
```
Comment (utsabc) [inline v0/destinations/hs/util.js:217]:
```suggestion
  return get(message, `${sourceKey}`)[`${lookupField}`];
```

**Review by utsabc — APPROVED** [review]

File: v0/destinations/hs/util.js [hunk 3]
```diff
@@ -215,45 +252,18 @@ const searchContacts = async (message, destination) => {
...
-    traits[`${propertyName}`];
-
-  if (!value) {
+  const lookupFieldInfo =
+    getLookupFieldValue(message, Config.lookupField) ||
+    getLookupFieldValue(message, "email");
+  if (!lookupFieldInfo?.value) {
     throw new CustomError(
-      `[HS] Identify:: '${propertyName}' lookup field for contact lookup not found in traits`,
+      "[HS] Identify:: email i.e a deafult lookup field for contact lookup not found in traits",
```
Comment (ItsSudip) [inline v0/destinations/hs/util.js:266]:
```suggestion
      "[HS] Identify:: email i.e a default lookup field for contact lookup not found in traits",
```

**Review by utsabc — APPROVED** [review]

**Review by ItsSudip — APPROVED** [review]

**Pull request merged**